### DISCUSSION TITLE
NORA paid subscription feedback

### DISCUSSION THREAD

**Discussion by andrei-tatar** [andrei-tatar/node-red-contrib-smartnora#63] [General]
It's been a while since Smart NORA was launched and it's now generating a steady monthly cost for the infrastructure that Google provides

We're now at well over 2.3k registered users and ~1.5k online users staying connected each moment of the day.

I've been thinking of adding a paid subscription model to support the ongoing development and the infrastructure costs for NORA so this discussion is focused around feedback from users. As a payment service I was thinking of using [Paylike](https://paylike.io/pricing) (has a commission of 25 cents + 1.35%) so I think the subscription will have to be paid for a full year (or at least it will be cheaper) - feel free to recommend other payment services that you know and worked with.

There's also a pretty big overhead because I would need to start a business in order to accept payments (meaning taxes/accountant services/etc.).

There will still be the possibility to use a free account with a limit on the number of devices you can sink over with Google (haven't decided on the threshold yet).

So, here are the main questions:
- how satisfied are you with NORA so far?
- would you pay for a monthly subscription to use Smart NORA?
- how much would you pay for this service? (paid monthly vs paid yearly)
- what features do you think are missing from Smart NORA?
- would you like other integrations for Smart NORA (other than Node-RED)?

**Comment by rgerrans** [comment]
- how satisfied are you with NORA so far?
Very
- would you pay for a monthly subscription to use Smart NORA?
Yes
- how much would you pay for this service? (paid monthly vs paid yearly)
$25-50 per year seems reasonable given all the benefits
- what features do you think are missing from Smart NORA?
None that I can think of at this point
- would you like other integrations for Smart NORA (other than Node-RED)?
No, I'm good with Node-Red since that's the heart of my control logic

**Comment by ma261065** [comment]
how satisfied are you with NORA so far?
It's very good
would you pay for a monthly subscription to use Smart NORA?
Yes
how much would you pay for this service? (paid monthly vs paid yearly)
$1 to $2 a month would seem reasonable. Or if offering a free service, I'd limit it to 10 devices.
what features do you think are missing from Smart NORA?
It's excellent, but I haven't used it enough to find any deficiencies
would you like other integrations for Smart NORA (other than Node-RED)?
I only use Node-RED, so I'm ok with it as it is

**Comment by KSumwalt** [comment]
- how satisfied are you with NORA so far?
  - Very
- would you pay for a monthly subscription to use Smart NORA?
  - Yes
- how much would you pay for this service? (paid monthly vs paid yearly)
  - A, a monthly fee of $2-5 with an option of a yearly payment discounted seems fair to me.  I agree with the free up to X devices as well.  Or at least a free 30 day trial.
- what features do you think are missing from Smart NORA?
  - Media Remote, the Streaming devices
- would you like other integrations for Smart NORA (other than Node-RED)?
  - Node Red is enough for me, thank you!

**Comment by storm1er** [comment]
> how satisfied are you with NORA so far?

Very

> would you pay for a monthly subscription to use Smart NORA?

Yes, but would prefer yearly

> how much would you pay for this service? (paid monthly vs paid yearly)

- freemium for one device (allowing people to test)
- up to 10 devices
  - 2 to 3€/month
  - 20 to 30€/year
- unlimited devices
  - 3 to 6€/month
  - 30 to 50€/year

> what features do you think are missing from Smart NORA?

Camera / Streaming devices

> would you like other integrations for Smart NORA (other than Node-RED)?

Thinking about [Domoticz](https://www.domoticz.com/) & [Gladys](https://gladysassistant.com/)
I know some blind people who use both, could be helpful for them
Personally, node-red is enough =)

**Comment by patrik-malina** [comment]
how satisfied are you with NORA so far?
Very

would you pay for a monthly subscription to use Smart NORA?
yes, prefer yearly tho.

how much would you pay for this service? (paid monthly vs paid yearly)
about 2-3€ monthly, 20-30€ yearly. Maybe more, idk..

what features do you think are missing from Smart NORA?
maybe security, but i is see its working on it.

would you like other integrations for Smart NORA (other than Node-RED)?
i'm fine

**Comment by Dud3z** [comment]
how satisfied are you with NORA so far?
very

would you pay for a monthly subscription to use Smart NORA?
if the price isnt to high

how much would you pay for this service? (paid monthly vs paid yearly)
i was registered at another service and paid for it. it was 2usd or euro a month.
yearly the equivalent of 11 or 10 months.

what features do you think are missing from Smart NORA?
maybe a camera/Nest integration?

would you like other integrations for Smart NORA (other than Node-RED)?
nope

**Comment by lucavivan** [comment]
-   **how satisfied are you with NORA so far?**
  I'm really happy

- **would you pay for a monthly subscription to use Smart NORA?**
 Yes

- **how much would you pay for this service? (paid monthly vs paid yearly)**
$25-30 per year unlimited devices

-  **what features do you think are missing from Smart NORA?**
other devices like window, security system etc... (not really important because the major devices are already supported) 

- **would you like other integrations for Smart NORA (other than Node-RED)?**
no

**Comment by Toby-Mills** [comment]
**how satisfied are you with NORA so far?**
- very

**would you pay for a monthly subscription to use Smart NORA?**
- yes

**how much would you pay for this service? (paid monthly vs paid yearly)**
- $15 per year (maybe charge more for large number of devices?)

**what features do you think are missing from Smart NORA?**
- would like to open & close doors without needing a PIN, but I suspect that's a Google issue

**would you like other integrations for Smart NORA (other than Node-RED)?**
- Node-RED is all I need

**Comment by EzequielPitton78** [comment]
**how satisfied are you with NORA so far?**

- very

**would you pay for a monthly subscription to use Smart NORA?**

- yes

**how much would you pay for this service? (paid monthly vs paid yearly)**

- $20-30 per year

**what features do you think are missing from Smart NORA?**
 

- Nothing important, maybe more devices support & more speed/stability in case there is potential.

**would you like other integrations for Smart NORA (other than Node-RED)?**

- Not really, Node-RED is perfect

**Reply by andrei-tatar** [reply]
> Nothing important, maybe more devices support & more speed/stability in case there is potential.

Device support is always in the work. On the speed & stability: if you experience issues, you can always open an issue on github to discuss them. If it's related to NORA, should be fixable. If related to Google Assistant, nothing we can do :(

**Comment by sonar98** [comment]
- **how satisfied are you with NORA so far?**
Very
- **would you pay for a monthly subscription to use Smart NORA?**
Yes
- **how much would you pay for this service? (paid monthly vs paid yearly)**
Would like a license model with free up to X devices (maybe 10?). Paid license with discount for yearly subscription (pay 10 months get 12) 1$ or 2$ a month.
- **what features do you think are missing from Smart NORA?**
Doorbell, curtain, smoke detector, co detector, streaming box/stick etc. (and other devices supported by google smart home 😉 
- **would you like other integrations for Smart NORA (other than Node-RED)?**
Direct Domoticz integration

**Comment by bjuedes** [comment]
how satisfied are you with NORA so far?
- Very, local execution is great

would you pay for a monthly subscription to use Smart NORA?
- Yes

how much would you pay for this service? (paid monthly vs paid yearly)
- approx. 20 € per year is OK

what features do you think are missing from Smart NORA?
- so far the most common are available and I think more will come

would you like other integrations for Smart NORA (other than Node-RED)?
- no, only Node-Red in place

**Comment by andrei-tatar** [comment]
So I integrated stripe for subscriptions. There are 2 prices available: 2 eur / month and 20 eur / year.
If/When the number of subscriptions cover the business overhead, the price will get lower.
I don't want to make a living out of NORA just covering the costs and a few beers :)

After 1st December, users with no subscriptions will only get 5 devices, across all groups, synced with Google Home.

**Reply by Dud3z** [reply]
Paid, but the sub is not showing up.

**Reply by andrei-tatar** [reply]
@Dud3z Interesting, can you provide more details? What is your uid, what do you see when going to https://smart-nora.eu/my-nora

**Reply by andrei-tatar** [reply]
@Dud3z I fixed it. It was a problem with the caching because of the service worker.

**Reply by Toby-Mills** [reply]
Paid - thanks for the service :)

**Reply by EzequielPitton78** [reply]
Paid, thank you too for the service!!

**Reply by patrik-malina** [reply]
Paid also

**Reply by normanr** [reply]
Paid. I was surprised to find out that I'm already at 8 devices! (and have plans that will easily double that)

**Reply by bjuedes** [reply]
Hi Andrei, already paid a couple of days ago. But so far it's not visible in my account - do I miss anything?
Can you check this pls?

**Reply by andrei-tatar** [reply]
@bjuedes I can but I will need your used id. It's displayed when you visit: https://smart-nora.eu/my-nora

**Comment by JasonJBloom** [comment]
Paid, thank you for such a great service.

**Comment by bjuedes** [comment]
Hi Andrei,

my ID is H1yupFtCQ7ePFZZkgNzvnOpta9s2, thanks!

 

BR Bernd

 

Von: Andrei Tatar ***@***.***> 
Gesendet: Dienstag, 19. Oktober 2021 08:51
An: andrei-tatar/node-red-contrib-smartnora ***@***.***>
Cc: bjuedes ***@***.***>; Mention ***@***.***>
Betreff: Re: [andrei-tatar/node-red-contrib-smartnora] NORA paid subscription feedback (Discussion #63)

 

@bjuedes <https://github.com/bjuedes>  I can but I will need your used id. It's displayed when you visit: https://smart-nora.eu/my-nora

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/andrei-tatar/node-red-contrib-smartnora/discussions/63#discussioncomment-1500255> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AUAJQXKANOYJRY4G66VPLXDUHUINRANCNFSM5A753AUA> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Reply by andrei-tatar** [reply]
@bjuedes Stripe says the payment failed. I sent you an email with the details.

**Comment by bjuedes** [comment]
Okay, I will check and pay again then.

 

Thanks!

 

BR Bernd

 

Von: Andrei Tatar ***@***.***> 
Gesendet: Dienstag, 19. Oktober 2021 09:19
An: andrei-tatar/node-red-contrib-smartnora ***@***.***>
Cc: bjuedes ***@***.***>; Mention ***@***.***>
Betreff: Re: [andrei-tatar/node-red-contrib-smartnora] NORA paid subscription feedback (Discussion #63)

 

@bjuedes <https://github.com/bjuedes>  Stripe says the payment failed. I sent you an email with the details.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/andrei-tatar/node-red-contrib-smartnora/discussions/63#discussioncomment-1500430> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AUAJQXKLKDVSDFYGLN27Y2TUHULXBANCNFSM5A753AUA> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Comment by Wireheadbe** [comment]
Paid - definitely worth it!
edit: maybe put it in the Readme.md ?

**Comment by blackest** [comment]
ok today all but 3 lights stopped working with smart nora , but thats ok with a 12 euro echo flex I still have control. I will be honest here and say i have found smart nora to work quite well  until one light stopped turning on about a month a go. Not a hope in hell that I will pay a subscription to turn my lights on.  When there are free alternatives. I prefer local tuya to the tuya clould app and google is a maniac turning of sockets in a room as well as the light. Which is why i didn't want to use the tuya integration in google home. 

I could see myself making a one of payment for smart nora but never a subscription. There are light switches after all.

**Reply by andrei-tatar** [reply]
Not a problem. Thanks for the feedback!

**Reply by andrei-tatar** [reply]
On the light that doesn't work anymore, if you still use the service, you can open an issue with the details. I suspect it's not Nora related, but you never know.

**Reply by blackest** [reply]
The light that stopped working only stopped turning on , it still worked to turn it off  and the integration was identical for the other 8 tuya lights, The node says API Error I can still turn the light on with alexa and the same local tuya integration. So it's looking to be Nora or maybe node red that had the problem. 

It's a good project but subscribing to turn my lights on.  It also defeats the purpose of having smart bulbs which turn off automatically.

**Comment by y8s** [comment]
I see a lot of pro paid user feedback.
Nora/Smartnora is a great service and worth the cost for those willing to pay. I've used both iterations with great success for a long time. I get why it is popular and why you have to charge.

But the recent abrupt crippling of the free version was a tragic event for me. I've lost most of my google integration and have had to find workarounds for it. Like I said: I get why you had to charge and that's fine. My issue is that both the original Nora and later Smartnora were provided initially without restriction and then feature-reduced for free users. Had there been a hard limit initially with a clear explanation that the service was freemium, I would have understood and made my choice at the outset. This is also exactly what happend with services like IFTTT and Pushbullet not too long ago. When they crippled their free service after offering it for so long, I left and found other solutions.

I run my smart home on a shoestring budget and really want to avoid recurring costs. To that end, rather than subscribe, I found the "other" google assistant nodered integration and created my own google app. I suppose this reduces your user load by 1 and puts the work on me to get the service up and running. I did that and it works so far.

Eventually I'll migrate the rest of my google assistant nodes over to my own app's nodes and disable Smartnora. 

I do want to share my appreciation for the service while I used it as well as for inspiring the work to get google assistant to work so well with nodered. Thanks for that.

**Reply by andrei-tatar** [reply]
Thanks for the feedback!

I understand, the problem is you can't provide a service, with ongoing connection, to thousands of users, for free, for a very long time (unless you are doing something else to pay for the cloud services - ads/collecting and selling data/etc).

This is the disclaimer I added to the login page of the first NORA service, when announcing the new Smart NORA:

> Being a big project hosted in the cloud, it will eventually require a subscription (probably around 2-3 eur/month) but it will always offer a free account type (probably with a limit on the number of devices).

**Reply by blackest** [reply]
> I run my smart home on a shoestring budget and really want to avoid recurring costs. To that end, rather than subscribe, I found the "other" google assistant nodered integration and created my own google app. 

I am quite interested in what the other google assistant integration  could be. In the long term it may be the least limiting option. Alexa isn't too bad for a voice assistant, but google assistant is a bit more flexible. E.g with the "local Bedroom Lamp"  Alexa will work,  Google will also work with variations such as bedroom light , bedroom lamp, turn on the light I am thinking of (requires mini in the same room).  Alexa is far more brief in replies "OK" but will seek clarification when needed. All the assistants seem to be  little flawed and as for working together ... 

Thank you for your project Andrei, I hope you don't mind some talk of alternatives, i'm sure there are a lot of people considering where to go from here.

**Reply by andrei-tatar** [reply]
@blackest  I don't mind, but I think there are better places for that discussion: https://discourse.nodered.org/

**Reply by y8s** [reply]
@blackest It's this one: https://flows.nodered.org/node/node-red-contrib-google-smarthome
large barrier to entry but does do what you expect.  

I couldn't figure out a way to send a private message to blackest so I put it here. If you want this removed, that's totally cool.

**Reply by Armageddit** [reply]
> @blackest It's this one: https://flows.nodered.org/node/node-red-contrib-google-smarthome large barrier to entry but does do what you expect.
> 
> I couldn't figure out a way to send a private message to blackest so I put it here. If you want this removed, that's totally cool.

i do not want to use 3rd party providers and certainly not their servers!
I thought smart-nora only connects directly to Google... actually I have not really thought about it...
i can just as well access the dashboard of node-red via VPN.
but sometimes it is more comfortable to just talk
so now i will look at this alternative and not use smart nora anymore

thanks y8s!

too bad but for that little bit i don't pay 2€/month



Translated with www.DeepL.com/Translator (free version)

**Comment by markusoliverklein** [comment]
I would like to take out a subscription, but unfortunately I am missing some important information about it.
How can I cancel a subscription?
When do I have to cancel the subscription?

Maybe it's written somewhere and I just can't find it :-) 

Thanks for the information.

Regards

Markus

**Reply by andrei-tatar** [reply]
Once you create a subscription, you can cancel/change it at any time from https://smart-nora.eu/my-nora .

You will see a `Manage subscription` button that will take you to Stripe Customer Portal where you can cancel, change the plan (monthly vs yearly), change billing information, etc.

Canceling a subscription maintains it active until the end of the paid period, it's not immediate.

Good observation, I'll probably have to write up some doc about it on the main readme page on github.

**Reply by markusoliverklein** [reply]
Subscription done.

Thank you for the fast response.

**Reply by andrei-tatar** [reply]
Thanks for the support! Updated the readme with the info.

**Comment by Howie55** [comment]
Hi Andrei, got caught out when 22 of my 27 devices disappeared from Google Assistant until I logged in to the Smart NORA page and noticed that it is now a subscription service.  Paid my dues and then re-introduced all the devices back into Google Assistant.  Been a user since the original NORA days and the World is becoming a user pays society so this explains the way forward.  I have noticed previously when the node is updated that I sometimes have to re-introduce the devices back as they don't seem to always remember which room they have been assigned to.  Am I the only one who has had this issue?

Howard.

**Reply by andrei-tatar** [reply]
@Howie55 Hi, I suspect you changed the rooms from your Google Home app without updating them in node-red as well. The changes from Google Home do not sync back into node-red.

**Comment by Syridian** [comment]
_1. How satisfied are you with NORA so far?_
       Up until my home stopped working because I don't check this page when everything is working fine and I didn't seem to get a notification via email that the service was going to become a subscription server, I was quite satisfied.  To be honest, I still am, I'm more of a user than a developer, so I'm not continually keeping up to date and usually only check when things stop working.  As a paid service, I would expect notifications about my subscription status.
       
_2. Would you pay for a monthly subscription to use Smart NORA?_
       Most definitely, and have now done so.  
       
_3. How much would you pay for this service? (paid monthly vs paid yearly)_
       About what I paid for the current level of service.  I would prefer a yearly cost as it's just a one-off payment.  Though I would also prefer better customer care if I'm paying for the service, like automated emails informing me that the service would change on a certain day if I don't make sure my subscription is up to date.  I have checked my email and could not find anything even though I do have other emails from Smart NORA.  
       
_4. What features do you think are missing from Smart NORA?_
       A Switch node that bypasses Googles Scheduling limitations on Switches, as I use switches for other things such as voice control of configuration settings and being able to schedule those would be good.  There are ways around it, but they get messy as they end up adding lights to Google Home that don't exist, and Google's way of grouping lights means that that gets very messy quite fast.

The ability to identify which Google device has initiated a query so that any responses generated by node-red can be sent back to that specific device

The ability to either hide lights, or group lights in node-red so that they only appear as a single light in Google Home.
       
_5. Would you like other integrations for Smart NORA (other than Node-RED)?_
      Don't think I would use them, so nothing comes to mind.

**Reply by andrei-tatar** [reply]
> As a paid service, I would expect notifications about my subscription status.

You should be getting email notifications for the subscription. I did send a notification when the subscription was added. Next big change will be notified by email.

> I would prefer a yearly cost as it's just a one-off payment.

There is the option for an yearly payment (20 eur/year - one time payment or 2eur/month - monthly payment).

>  I do have other emails from Smart NORA.

The only email should be the "verify e-mail address" email.

> The ability to identify which Google device has initiated a query so that any responses generated by node-red can be sent back to that specific device

Can you please detail?

> The ability to either hide lights, or group lights in node-red so that they only appear as a single light in Google Home.

It's not possible to tell Google Home how to group lights. But you could use a single NORA light and control multiple real lights.

**Reply by Syridian** [reply]
Hi Andrei,

Thanks for your reply.  

Here's the detail for "The ability to identify which Google device has initiated a query so that any responses generated by node-red can be sent back to that specific device".

Like a lot of people, I have multiple Google Home devices around the house.  I would like to be able ask Google Home to perform an action and have Node-Red respond to that at a later stage when the action is done.  At the moment I can send notifications to specific Google Home devices, but I have no way of knowing where that task was initiated.  The simple workaround for this is to send notifications to every device, but if I had the initiating node identify the name of the device (say in the topic) that triggered that sequence, then I can return the response directly to where it was initiated without disturbing the rest of the house. 

An example of this is that I have a smart kettle that I can turn on from any room that has a Google Home device.  Node-red monitors the temp of the kettle and alerts me when the kettle has boiled as well as if I have left it too long and forgotten about boiling it.  Currently, it only alerts me in one room, so if I'm in another room I can't hear the notification from the other room, and the notification may be irrelevant or annoying to the person in the room that gets that notification.  Being able to direct that response back to where the initial voice command was asked is a much neater way of doing things and will allow me to use Smart NORA in a whole new way.

**Reply by andrei-tatar** [reply]
I agree, that would be cool. But I'm limited by what Google provides in the API and it does not provide any kind of "initiating device" identifier.

**Reply by normanr** [reply]
The API does allow delayed responses (that are returned to the device that initiated the request). It must be sent within 5 minutes though. More information at https://developers.google.com/assistant/smarthome/develop/notifications

**Reply by normanr** [reply]
For grouping of lights, note that you can turn all lights in a room on/off at once already. "Turn on/off the $room lights". If you need more specific control then maybe look at Scenes?

**Reply by Syridian** [reply]
Hi @normanr,

My usage case doesn't require the API to allow for delayed responses as the responses are generated within Node-red and sent to the speaker via a cast.  Hence hoping there was some way of identifying the initiating device.  I'm trying to keep as much as possible in Node-red rather than having it spread across multiple systems that need to interact.

The grouping issue is not really an issue, it was just an answer to Andrei asking if there were features we would like to see.  It's Google's problem, not Smart-Nora's.

**Reply by normanr** [reply]
It looks like the API for "follow-up responses" gives a followUpToken that can be used to reply to the same device that the action was executed from. I was hoping that the async command execution would support that, but I don't think it does currently (there seems to be a 1500ms limit). It would be very cool if async command execution supported a way to return PENDING, and then later SUCCESS. Follow-up responses are only be supported for `LockUnlock`, `TestNetworkSpeed`, and `OpenClose` traits (unclear about `ArmDisarm`), so that probably doesn't work for you.

Proactive notifications are supported for `ObjectDetection`, `RunCycle`, and `SensorState` traits, but those don't model to "you haven't used the hot water yet" very well (and would be announced to all devices anyways).

**Reply by normanr** [reply]
I realized that it should be possible to provide this for locally fulfilled requests, so I've filled https://github.com/andrei-tatar/node-red-contrib-smartnora/issues/188 for it.

**Comment by Anjerlaan** [comment]
Hey Adrei,

My subscription was not extended automatically because my credit card is new.
Now I have selected Paypal instead of Visa Card, but my smart-nora page is still telling me that I have no active subscriptions?

Can you check if you have received my supscription via Paypal?

Greetzzz,

Gerben Schmidt

**Reply by andrei-tatar** [reply]
Hey @Anjerlaan ,
Replied over email.
